searchbox.py

```diff
@@ -0,0 +1,202 @@
+
+    Example: format_county('SAN FRANCISCO') returns 'San Francisco'
+    '''
+    res = ''
+    county_words = selected_county.split(' ')
+    for word in county_words:
+        res += word.lower().capitalize() + ' '
+    return res[:-1]
+
+# Create some session state variables to track user interaction
+if "first_time" not in st.session_state: # If this is the first time loading the script, track that
+	  st.session_state["first_time"] = True 
+if "county_list" not in st.session_state: # If we haven't loaded county names before, get ready to
+	  st.session_state["county_list"] = None
+
+# Initial data load
+@st.cache_data
+def load_county_list():
+    '''
+    Load data from state_counties_csv on first time running the app
+    '''
+    # Get list of counties in the format COUNTY, STATE
+    url = "https://raw.githubusercontent.com/edgi-govdata-archiving/"
+    url += "ECHO_modules/packaging/data/state_counties_corrected.csv"
+    all_counties = pd.read_csv( url )
+    # Data pre-processing - make sure the only states in the list are real states, but comparing with the fips data
+    all_counties = all_counties.loc[all_counties["FAC_STATE"].isin(state_abbreviations)]
+
+    column_1 = 'County'
+    column_2 = 'FAC_STATE'
+    # combine county and state columns
+    all_counties['County_Pairs'] = all_counties[column_1] + ', ' + all_counties[column_2]
+    county_list = all_counties['County_Pairs']
+    # filter duplicates
+    county_list = county_list.unique()
+
+    return county_list
+
+## Only load counties if this is the first run through of the script
+if st.session_state["first_time"]:
+    st.session_state["county_list"] = load_county_list()
+    st.session_state["first_time"] = False
+
+# Let user select county
+user_selection = st.selectbox(
+    'Search for a county',
+    st.session_state["county_list"]
+)
+
+comma_pos = user_selection.find(',')
+
+# Extract state and county from user's selection
+selected_state = user_selection[comma_pos + 2:]
+selected_county = user_selection[:comma_pos]
+
+county_name = format_county(selected_county)
+
+# handle errors with empty columns
+try:
+    st.markdown('### Get Facilities in Selected County')
+    # Get county names based on Steve's work.
+    counties = pd.read_csv("https://raw.githubusercontent.com/edgi-govdata-archiving/ECHO_modules/main/data/state_counties_corrected.csv") # Get a county name lookup
+    counties = counties.groupby(by=["FAC_STATE", "County", "FAC_COUNTY"]).count() # Restructure the data
+    counties = counties.iloc[(counties.index.get_level_values('County') == selected_county) & (counties.index.get_level_values('FAC_STATE') == selected_state)].reset_index() # Search for COUNTY, STATE
+    counties = list(counties["FAC_COUNTY"].unique()) # List FAC_COUNTY names in ECHO associated with COUNTY
+    # Format a sql query
+    c = "("
+    for county in counties:
+      c += "\'"+county+"\',"
+    c = c[:-1] + ")"
```

Comment:
this is forwarded feedback from Jacob -- basically saying: 
I think it would be nice to separate function based code vs script based code. For instance in searchbox.py or get_sbu_data.py there is a few function definitions, but then there's also just blocks of code (doing things like sql queries). I think the functions should be stored in another file, and then called from this one. Overall, I think that will make the file a little bit more readable

Reply:
I think also, in the future, we would want to separate the streamlit component-creating code and the data-retrieving code